### DISCUSSION TITLE
@auth/core adapter

### DISCUSSION THREAD

**Discussion by rolanday** [remix-run/remix#6583] [Proposals]
Can we please get an officially supported next-auth adapter for Remix? Remix-auth is an excellent contribution, and it works, but not having much luck getting answers to questions,the project appears to be best-effort, and samples don't go much beyond what is useful for hobby projects. Auth is an area that developers shouldn't be left to their own devices to implement or finish, and svelte, solid, and qwik all include next-auth adapters since before v1.0. Clerk provides a solution; however, it can get expensive and lacks the sort of tight integration that next-auth can provide. IMO, this is area that Remix is sorely lacking and should require relatively minimal effort for core maintainers to implement.

**Comment by MichaelDeBoey** [comment]
`next-auth` is a Next.js related project, no?

If you have any questions about `remix-auth`, @sergiodxa can probably help you out.

**Reply by rolanday** [reply]
@MichaelDeBoey , it used to be the case that next-auth was bound to nextjs, but this is no longer the case w/ auth.js (https://authjs.dev, a.k.a., @auth/core), which provides next-auth core for other non-next frameworks to use (e.g., qwik, solid, and svelte). I still fall into the trap of referring to auth.js as next-auth out of habit so apologies for any confusion. One person has even posted a [gist](https://gist.github.com/lawrencecchen/75fee3be74a8cc56aedae66b17d6d83c) applying next-auth to remix, but that's not a rabbit hole I'm interest in going down.

Remix-auth is working out very well for me and I prefer it in many ways to next-auth / @auth/core, but I needed to put in quite a bit more effort as compared to other projects using @auth/core, including forking due to lack of responses for help on discord (presumably because no good answer), and, I'd feel a lot better using @auth/core since fully featured and actively maintained. I thank @sergiodxa for the contrib as Remix would have been a non-starter for me otherwise, and Remix has turned out to be my fav framework to use and I am overall exceedingly happy with Remix. Still, it's curious to me that the Remix team leaves a capability (auth) so fundamental to securing a real-world website as 3p contrib.

P.S. updated title to read @auth/core adapter to prevent any further confusion.

**Reply by MichaelDeBoey** [reply]
I don't think auth is a core thing the team currently wants to focus on (correct me if I'm wrong @brophdawg11 @jacob-ebey @pcattori)

As this is a third party lib, I would suggest to either try to fix your problems with @sergiodxa's `remix-auth` _or_ to contribute the gist to the `@auth/core` project to have it maintained by their team

**Reply by rolanday** [reply]
@MichaelDeBoey, I am not having any issues getting remix-auth to work for me -- I feel I made that very clear. I am simply making a feature suggestion because I'd like to see Remix continue to be successful and grow, and auth is an area Remix is sorely lacking IMO (even w/ remix-auth), giving developers one more reason to choose Next over Remix -- and addressable by implementing an adapter for an established auth library, has been done for other frameworks -- i.e., doesn't need to be a core thing the core Remix team focuses on. I am not the right person to implement such an adapter, even if I had the time.
